staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -59,8 +59,12 @@ func StorageWithCacher(capacity int) generic.StorageDecorator {
 
 		// TODO: we would change this later to make storage always have cacher and hide low level KV layer inside.
 		// Currently it has two layers of same storage interface -- cacher and low level kv.
+		cacheCapacity := 0
+		if capacity != nil {
+			cacheCapacity = *capacity
+		}
 		cacherConfig := cacherstorage.Config{
-			CacheCapacity:  capacity,
+			CacheCapacity:  cacheCapacity,
```

Comment:
If I read related code correctly, it won't truly respect resource size configured in `--wach-cache-sizes` since cacher's CacheCapacity deprecated, it means `capacity *int` actually makes no sense.
IMO if we want to respect resource capacity configured by `--wach-cache-sizes`, CacheCapacity should be used as initial capacity which users want to set.
If we only want `--wach-cache-sizes`  switch off cache for individual resources, can size of resource be ignored?

Reply:
The main point of leaving the `--watc-cache-sizes` for now is that it is the only way for now that allows us to switch on or off watch cache for individual resources.

Reply:
Adjusted your suggestion, but that makes sense.